pvlib/inverter.py

```diff
@@ -16,6 +16,39 @@
 from numpy.polynomial.polynomial import polyfit  # different than np.polyfit
 
 
+def _sandia_eff(v_dc, p_dc, inverter):
+    r'''
+    Calculate the inverter AC power without clipping
+    '''
+    Paco = inverter['Paco']
+    Pdco = inverter['Pdco']
+    Vdco = inverter['Vdco']
+    C0 = inverter['C0']
+    C1 = inverter['C1']
+    C2 = inverter['C2']
+    C3 = inverter['C3']
+    Pso = inverter['Pso']
+
+    A = Pdco * (1 + C1 * (v_dc - Vdco))
+    B = Pso * (1 + C2 * (v_dc - Vdco))
+    C = C0 * (1 + C3 * (v_dc - Vdco))
+
+    return (Paco / (A - B) - C * (A - B)) * (p_dc - B) + C * (p_dc - B)**2
+
+
+def _sandia_limits(power_ac, p_dc, Paco, Pnt, Pso):
+    r'''
+    Applies minimum and maximum power limits to `power_ac`
+    '''
+    power_ac = np.minimum(Paco, power_ac)
+    try:
+        power_ac[p_dc < Pso] = -1.0 * abs(Pnt)
+    except TypeError:  # float
+        if power_ac < Pso:
+            power_ac = -1.0 * abs(Pnt)
```

Comment:
I was going to say the `np.where` in the original function does all of this in one line, but the conditional in the array case is different from the conditional in the float case, so I'm just confused.

Reply:
Line 47 is a typo, should be `p_dc < Pso:`.  The `try/except` is an attempt to handle the case when `power_ac` is a float. `np.where` converts `Series` to `ndarray`, and I was hoping to avoid another use of `isinstance` to preserve input type. I'm OK using the technique in `inverter.sandia` if you prefer that.

Reply:
> np.where converts Series to ndarray, and I was hoping to avoid another use of isinstance to preserve input type.

Ah, I forgot about that. Perhaps your method is preferable because it works with any array-like input (e.g. dask). I suggest assigning the condition and the limit value to variables before the try/except block to ensure there's no inconsistency.